tardis/energy_input/tests/test_gamma_ray_channel.py

```diff
@@ -0,0 +1,235 @@
+import pytest
+import numpy as np
+from pathlib import Path
+import astropy.units as u
+import numpy.testing as npt
+import radioactivedecay as rd
+import astropy.constants as const
+from radioactivedecay import converters
+
+from tardis.model import SimulationState
+from tardis.io.configuration import config_reader
+from tardis.energy_input.energy_source import (
+    get_nuclear_lines_database,
+)
+from tardis.energy_input.gamma_ray_channel import (
+    create_isotope_dicts,
+    create_inventories_dict,
+    calculate_total_decays,
+    create_isotope_decay_df,
+)
+
+
+@pytest.fixture(scope="module")
+def gamma_ray_config(example_configuration_dir: Path):
+    """
+    Parameters
+    ----------
+    example_configuration_dir: Path to the configuration directory.
+
+    Returns
+    -------
+    Tardis configuration
+    """
+
+    yml_path = (
+        example_configuration_dir
+        / "tardis_configv1_density_exponential_nebular_multi_isotope.yml"
+    )
+
+    return config_reader.Configuration.from_yaml(yml_path)
+
+
+@pytest.fixture(scope="module")
+def gamma_ray_simulation_state(gamma_ray_config, atomic_dataset):
+    """
+    Parameters
+    ----------
+    gamma_ray_config: Tardis configuration
+    atomic_dataset: Tardis atomic-nuclear dataset
+
+    Returns
+    -------
+    Tardis simulation state
+    """
+
+    gamma_ray_config.model.structure.velocity.start = 1.0 * u.km / u.s
+    gamma_ray_config.model.structure.density.rho_0 = 5.0e2 * u.g / u.cm**3
+    gamma_ray_config.supernova.time_explosion = 150 * u.d
+
+    return SimulationState.from_config(
+        gamma_ray_config, atom_data=atomic_dataset
+    )
+
+
+@pytest.fixture(scope="module")
+def gamma_ray_model_state(gamma_ray_simulation_state):
```

Comment:
I don't think this is a good function name when it returns isotope abundance and masses. Perhaps `gamma_ray_test_composition`?